scripts/linters/general_purpose_linter.py

```diff
@@ -868,6 +871,39 @@ def check_newline_at_eof(self):
         return concurrent_task_utils.TaskResult(
             name, failed, error_messages, error_messages)
 
+    def check_extra_js_files(self):
+        """Checks if the changes made include extra js files in core
+        or extensions folder which are not specified in
+        build.JS_FILEPATHS_NOT_TO_BUILD.
+
+        Returns:
+            TaskResult. A TaskResult object representing the result of the lint
+            check.
+        """
+        name = 'Extra JS files'
+        error_messages = []
+        files_to_lint = self.all_filepaths
+        failed = False
+
+        for filepath in files_to_lint:
+            if filepath.startswith(('core/templates', 'extensions')) and (
```

Comment:
start with:

```
if not (filepath.endswith('.js') and not filepath.endswith('protractor.js')) :
    continue
```

i.e, the first check should be the main blocker which in this case is endswith .js

Reply:
Done.